helpers/parser.py

```diff
@@ -1,96 +1,174 @@
 from PyQt6.QtCore import Qt
-from PyQt6.QtWidgets import (QFrame, QHBoxLayout, QLabel, QStyle,
+from PyQt6.QtWidgets import (QApplication, QHBoxLayout, QLabel, QStyle,
                              QPushButton, QVBoxLayout, QWidget)
 
 from helpers import config
 
 
-class ParserWindow(QFrame):
-    auto_hide_menu = True
-    always_on_top = True
+class ParserWindow(QWidget):
+    content = None
+    menu_area = None
+    name = ""
```

Comment:
I dislike even having a default for this anywhere -- part of why I liked a system that enforced it. I might have to take a cleanup pass on this whole thing later, but I think it can wait until we've got the modular stuff closer to complete.

Reply:
Yeah passing it down from the instance requires us to have it set in the class so it can use it in it's setup init stuff for config and whatnot. We can tweak it later on for sure, big thing right now, this is testable and it works basically the same as before.

Reply:
Actually, I set the name there, because I listed everything that we set on the class at the top. My understanding is that everything listed up there is like a "global" between parent class and instance. From what I gather, having the name as a class field, when we set self.name in the instance, then call super, super then has the self.name reference that we set allowing us to get all the stuff that required self.name in the super window for free.